crates/router/src/core/payments/transformers.rs

```diff
@@ -267,16 +270,50 @@ where
                 services::ApplicationResponse::Form(form)
             } else {
                 let mut next_action_response = None;
-                if payment_intent.status == enums::IntentStatus::RequiresCustomerAction {
+
+                let bank_transfer_next_steps =
+                    if let Some(storage_models::enums::PaymentMethod::BankTransfer) =
+                        payment_attempt.payment_method
+                    {
+                        let bank_transfer_next_steps: Option<NextStepsRequirements> =
+                            payment_attempt
+                                .connector_metadata
+                                .to_owned()
+                                .map(|metadata| {
+                                    metadata
+                                        .parse_value("NextStepsRequirements")
+                                        .change_context(
+                                            errors::ApiErrorResponse::InternalServerError,
+                                        )
+                                        .attach_printable(
+                                            "Failed to parse the Value to NextRequirements struct",
+                                        )
+                                })
+                                .transpose()?;
+                        bank_transfer_next_steps
+                    } else {
+                        None
+                    };
+
+                if payment_intent.status == enums::IntentStatus::RequiresCustomerAction
```

Comment:
The condition here is `AND` or `OR`?

Reply:
The condition should be OR, the use case is :
If bank_transfer_next_steps is some, then whatever the intent status may be, we go with bank_transfer_details, if bank_transfer is none and if intent_status is requires_customer_action, we go with RedirectToUrl